tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -707,15 +705,30 @@ func (s *storeImpl) GetMany(ctx context.Context, ids []{{$singlePK.Type}}) ([]*{
     } else if !ok {
         return nil, nil, nil
     }
-    {{- end}}
-
-	conn, release, err := s.acquireConn(ctx, ops.GetMany, "{{.TrimmedType}}")
+    {{- else if .Obj.IsDirectlyScoped }}
+    scopeChecker := sac.GlobalAccessScopeChecker(ctx)
+	scopeTree, err := scopeChecker.EffectiveAccessScope(permissions.ResourceWithAccess{
+		Resource: targetResource,
+		Access:   storage.Access_READ_ACCESS,
+	})
 	if err != nil {
-	    return nil, nil, err
+        return nil, nil, err
 	}
-	defer release()
+    {{- if .Obj.IsClusterScope }}
+    sacQueryFilter, err = sac.BuildClusterLevelSACQueryFilter(scopeTree)
+    {{- else}}
+    sacQueryFilter, err = sac.BuildClusterNamespaceLevelSACQueryFilter(scopeTree)
+    {{- end }}
+	if err != nil {
+        return nil, nil, err
+	}
+    {{- end }}
```

Comment:
I have the feeling the SAC query filter extraction code is common to a few functions. Would it make sense to extract this code in a single function ?

Reply:
Yeah, I also thought about that but I think it's better to merge everything and then do a cleanup. If we merge #1773 then code could be even more simplified